### ISSUE TITLE
A11y_Azure Tools Storage Explorer_Table_Table Grid_Name Role Value: Incorrect Role is given for sorting button as image.

### ISSUE THREAD

**Issue by keerthiduvvuri** [microsoft/AzureStorageExplorer#7609]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.32.1(96)

### Regression From

NO

### Architecture

x64

### Storage Explorer Build Number

20231114.10

### Platform

macOS

### OS Version

MAC Version 13.6.1

### GitHub Tags:
#A11yMAS;#A11yTCS;#MAC;#DesktopApp;#A11ySev2;#BM_AzureToolsStorageExplorer_MAC_Dec2023;#Azure Tools Storage Explorer;#FTP;#Voiceover;#WCAG4.1.2;#Name Role Value;#AILimited;

### Environment Details:
Application: Microsoft Azure Storage Explore
version: 1.32.1 (96)
Screen Reader: Voice Over

### Steps to Reproduce

1. Launch Storage Explorer
2. Tab Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Tab Navigate to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Tab Navigate to "Tables" of the selected subscription and expand it.
5. Tab Navigate to table.
6. Turn on Voice Over and Observe that Incorrect Role is given for sorting button as image.

### Actual Experience

Incorrect Role is given for sorting button as image.

### Expected Experience

Role should be defined as Button for sorting button.

### User Impact:
Screen reader users will face difficulty if Incorrect Role is given for sorting button as image.

### Attachment:
![A11y_Azure Tools Storage Explorer_Table_Table Grid_Name Role Value](https://github.com/microsoft/AzureStorageExplorer/assets/98728920/c925870d-44e8-452f-99cf-5831a1d946e2)

**Comment by craxal** [comment]
@keerthiduvvuri Can you provide additional details? I can't focus the image, because the sorting image is not a focusable element. Only the grid itself is focusable, and the arrow keys change which cell has focus.

**Comment by keerthiduvvuri** [comment]
@Craxal in that grid there is a sorting functionality present and when user is pressing return key sorting functionality is triggering out because it is interactive and when we navigate with Voice over it is just announcing as image, Image is not a Role. The Role property should be defined as Sorting Button.

**Comment by craxal** [comment]
@keerthiduvvuri I'm still not clear on a number of points:
* How are you navigating to the image? That is not normally accessible with tab and arrow keys. Without knowing this, I am not able to reproduce this issue.
* The image itself is not an interactive element. The header cell should be interactive. It's not clear as to how you are interacting with the sorting mechanism. (Also, "Sorting button" is not a valid role, but "button" is a valid role.)
* This may be more of a question of your testing procedure, but why are you turning VoiceOver on after navigating to the table? Why wouldn't you have VoiceOver turned on from the beginning? Does that produce different results?

**Comment by Rama8345** [comment]
@craxal 

Here expected is that the role as button should be provide to the "Arrow" control present in the column header as when we are pressing "SPACE" key on it the sort functionality is performing.

**Comment by craxal** [comment]
@Rama8345 The button role should belong to the entire header content, not the image, because clicking anywhere in the header will sort the column.

**Comment by kupatkar99** [comment]
Verified the issue in below environment:
Azure Tools Storage Explorer: Version: 1.33.0-dev (96)
AzCopy Version: 10.22.2
Platform: macOS
Build Number: 20240111.7
Issue is not fixed. Please find added attachment for reference.
Could you please provide the build where the fixes are avilable
<img width="1440" alt="7609_Not Fixed" src="https://github.com/microsoft/AzureStorageExplorer/assets/98530069/8b00072f-da24-47b9-84ab-1a2cd468dd87">

**Comment by kupatkar99** [comment]
Verified the issue in below environment:
Azure Tools Storage Explorer: Version: 1.33.0-dev (96)
AzCopy Version: 10.22.2
Platform: macOS
Build Number: 20240111.7
Issue is not fixed. Please find added attachment for reference.
Could you please provide the build where the fixes are avilable
<img width="1440" alt="7609_Not Fixed" src="https://github.com/microsoft/AzureStorageExplorer/assets/98530069/8b00072f-da24-47b9-84ab-1a2cd468dd87">

**Comment by craxal** [comment]
@kupatkar99 Most of our explorers use the same data grid control, but not all explorers have been updated. Please verify again with another explorer, such as for tables or for file shares.

**Comment by kupatkar99** [comment]
Verified the issue in below environment:
Microsoft Azure Storage Explorer
version: 1.33.0-dev (96)
Issue is still repro'ing. please find added attachment for reference.
<img width="1440" alt="MicrosoftTeams-image (4)" src="https://github.com/microsoft/AzureStorageExplorer/assets/98530069/6c967198-5a0a-4ab0-b44e-82d90bce756b">

**Comment by craxal** [comment]
@kupatkar99 As I mentioned earlier, the image itself is not supposed to be a button. The entire contents of the header are clickable, and therefore it was given the button label.

**Comment by kupatkar99** [comment]
Verified the issue in below environment:
Microsoft Azure Storage Explorer
Version: 1.33.0 (97)
AzCopy Version: 10.23.0
Platform: macOS
Architecture: arm64
Build Number: 20240220.6
Issue is fixed. hence closing the bug please find added attachment for reference.
<img width="1440" alt="7609_fixed" src="https://github.com/microsoft/AzureStorageExplorer/assets/98530069/cacf4342-b505-422a-bae9-f4e26453ecdf">

**Comment by keerthiduvvuri** [comment]
##Closed;
